### ISSUE TITLE
Doesn't work with Vapor?

### ISSUE THREAD

**Issue by msnegurski** [Napp/xray-laravel#52]
Hey, curious if anyone encountered this issue. It just straight up doesn't work with Vapor. From doing some digging issue seems to be that HTTP_X_AMZN_TRACE_ID and LAMBDA_INVOCATION_CONTEXT variable are not present in either $_SERVER or $_ENV.

**Comment by msnegurski** [comment]
Xray is enabled and configured and general traces are present. But anything that happens inside Laravel app is absent. Here is a screenshot as example.

![CleanShot 2024-04-21 at 11 49 52@2x](https://github.com/Napp/xray-laravel/assets/6137647/c14df338-61f5-415a-badf-34d3cbfcb639)

**Comment by emaadali** [comment]
@msnegurski  Were you able to figure this out?

**Comment by msnegurski** [comment]
@emaadali unfortunately not, basically gave up on this idea

I made some custom trace id generation and got it working somewhat, however that means traces from Laravel will not be linked to traces automatically generated by Lambda (from screenshot)

**Comment by yani-** [comment]
Yes, it supports vapor out of the box. You have to configure the submitter in config/xray.php to:
```
    'submitter' => \Napp\Xray\Submission\DaemonSegmentSubmitter::class,
```

**Comment by phadaphunk** [comment]
Anyone made it work? Or is this dead. I can only see lambda invocations and http response codes but nothing else is being logged even though I followed the doc for vapor.
